src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -36,6 +42,201 @@ internal IntPtr[] GetWaitedHandleArray(int requiredCapacity)
             return _waitedHandles.Items;
         }
 
+        private void PlatformSpecificInitializeExistingThread()
+        {
+            _osHandle = GetOSHandleForCurrentThread();
+            _priority = MapFromOSPriority(Interop.mincore.GetThreadPriority(_osHandle));
+        }
+
+        private static SafeWaitHandle GetOSHandleForCurrentThread()
+        {
+            IntPtr currentProcHandle = Interop.mincore.GetCurrentProcess();
+            IntPtr currentThreadHandle = Interop.mincore.GetCurrentThread();
+            SafeWaitHandle threadHandle;
+
+            if (Interop.mincore.DuplicateHandle(currentProcHandle, currentThreadHandle, currentProcHandle,
+                out threadHandle, 0, false, (uint)Interop.Constants.DuplicateSameAccess))
+            {
+                return threadHandle;
+            }
+
+            // Throw an ApplicationException for compatibility with CoreCLR
+            var ex = new ApplicationException();
```

Comment:
ApplicationException constructor loads localized string from error message. It is a potentially a lot of code running that may stomp on LastWin32Error. Better to fetch the error code before that.